server/src/main/java/io/druid/server/coordination/ServerManager.java

```diff
@@ -262,7 +120,8 @@ public void dropSegment(final DataSegment segment) throws SegmentLoadingExceptio
     }
     String dataSourceName = getDataSourceName(dataSource);
 
-    final VersionedIntervalTimeline<String, ReferenceCountingSegment> timeline = dataSources.get(dataSourceName);
+    final VersionedIntervalTimeline<String, ReferenceCountingSegment> timeline = segmentManager.getDataSources()
```

Comment:
`getDataSources()` creates a copy, please add a method to SegmentManager to extract VersionedIntervalTimeline by dataSource.